File: bank-service/src/main/java/rs/edu/raf/service/CurrencyService.java
package rs.edu.raf.service;

import rs.edu.raf.dto.CurrencyCreateDTO;
import rs.edu.raf.dto.CurrencyDTO;

import java.util.List;

public interface CurrencyService {
    /**
     * Creates currency
     * @param currencyCreateDTO DTO which contains information about currency
     * @return {@link CurrencyDTO} object representing created currency
     */
    CurrencyDTO createCurrency(CurrencyCreateDTO currencyCreateDTO);

    /**
     * Gets all currencies
     * @return List of {@link CurrencyDTO} representing currencies
     */
    List<CurrencyDTO> getAllCurrencies();

    /**
     * Deletes currency
     * @param id the id of the currency to be deleted
     */
    void deleteCurrency(Long id);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/service/CurrencyService.kt
package org.example.bankservicekotlin.service

import org.example.bankservicekotlin.dto.CurrencyCreateDTO
import org.example.bankservicekotlin.dto.CurrencyDTO

interface CurrencyService {
    /**
     * Creates currency
     * @param currencyCreateDTO DTO which contains information about currency
     * @return [CurrencyDTO] object representing created currency
     */
    fun createCurrency(currencyCreateDTO: CurrencyCreateDTO): CurrencyDTO

    /**
     * Gets all currencies
     * @return List of [CurrencyDTO] representing currencies
     */
    fun getAllCurrencies(): List<CurrencyDTO>

    /**
     * Deletes currency
     * @param id the id of the currency to be deleted
     */
    fun deleteCurrency(id: Long)
}